homeassistant/components/reolink/__init__.py

```diff
@@ -77,23 +78,21 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
     async def async_device_config_update() -> None:
         """Update the host state cache and renew the ONVIF-subscription."""
-        async with async_timeout.timeout(host.api.timeout):
+        async with async_timeout.timeout(host.api.timeout * (RETRY_ATTEMPTS + 2)):
```

Comment:
+2?

Reply:
Inside the reolink_aio library the request is tried `RETRY_ATTEMPTS` (=3) with a timeout of `host.api.timeout`, if it then fails it will disconnect from the camera, invalidade the tokens and unsubscribe from the ONVIF connections. This unsubscribing needs two additional request that are tried only once with a host.api.timeout, so that is the +2.

Furthermore the timeouts schould be handeled properly in the upstream library so this `async_timeout.timeout` inside HA schould never be reached. However in the past a few times complex unexpected issues have resulted in mutex problems that can hang the request for multiple minutes. This `async_timeout.timeout` is a falesafe for those kind of unexpected issues, (that have all been solved by now by my knoladge, but of course there can always be unknown bugs).